bundles/org.openhab.binding.mercedesme/src/main/java/org/openhab/binding/mercedesme/internal/api/RestApi.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mercedesme.internal.api;
+
+import static org.openhab.binding.mercedesme.internal.Constants.*;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.openhab.binding.mercedesme.internal.Constants;
+import org.openhab.binding.mercedesme.internal.config.AccountConfiguration;
+import org.openhab.binding.mercedesme.internal.exception.MercedesMeApiException;
+import org.openhab.binding.mercedesme.internal.utils.Utils;
+import org.openhab.core.auth.client.oauth2.AccessTokenRefreshListener;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.storage.Storage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.daimler.mbcarkit.proto.VehicleEvents;
+import com.daimler.mbcarkit.proto.VehicleEvents.VEPUpdate;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+/**
+ * {@link RestApi} for single calls towards Mercedes servers
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class RestApi extends Authorization {
+    private final Logger logger = LoggerFactory.getLogger(RestApi.class);
+
+    private String commandCapabilitiesEndpoint = "/v1/vehicle/%s/capabilities/commands";
+    private String vehicleAttributesEndpoint = "/v1/vehicle/%s/vehicleattributes";
+    private String capabilitiesEndpoint = "/v1/vehicle/%s/capabilities";
+    private String poiEndpoint = "/v1/vehicle/%s/route";
+
+    public RestApi(AccessTokenRefreshListener atrl, HttpClient hc, AccountConfiguration ac, LocaleProvider l,
+            Storage<String> store) {
+        super(atrl, hc, ac, l, store);
+    }
+
+    public void restSendPoi(String vin, JSONObject poi) {
+        String poiUrl = Utils.getRestAPIServer(config.region) + String.format(poiEndpoint, vin);
+        Request poiRequest = httpClient.POST(poiUrl);
+        addBasicHeaders(poiRequest);
+        addAuthHeaders(poiRequest);
+        poiRequest.header(HttpHeader.CONTENT_TYPE, "application/json");
+        poiRequest.content(new StringContentProvider(poi.toString(), "utf-8"));
```

Comment:
This will automatically provide the header:
```suggestion
        poiRequest.content(new StringContentProvider(poi.toString(), "utf-8"), "application/json");
```

Reply:
resolved all agent and content types